### PR TITLE
Fix two issues in cherry_pick_pull.sh

### PR DESCRIPTION
**Pull request by zmerlynn**
- Ignore errexit in trap handler
- Check for branch (since two people have complained now..)

### CODE REVIEW

**Comment by zmerlynn** [comment]
Over to @thockin for complaining. :)

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -29,7 +29,8 @@ declare -r STARTINGBRANCH=$(git symbolic-ref --short HEAD)
 declare -r REBASEMAGIC="${KUBE_ROOT}/.git/rebase-apply"
 
 if [[ "$#" -ne 2 ]]; then
-  echo "${0} <pr-number> <branch>: cherry pick <pr> onto <branch> and leave instructions for proposing pull request"
+  echo "${0} <pr-number> <remote branch>: cherry pick <pr> onto <remote branch> and leave instructions for proposing pull request"
+  echo "  - This script checks out <remote branch> and handles the cherry-pick of <pr> for you."
```
Comment (thockin) [inline hack/cherry_pick_pull.sh]: an "example: $0 12345 upstream/release-3.14" would help

**Comment by thockin** [comment]
One nit then LGTM

**Comment by zmerlynn** [comment]
Fixed.

**Comment by zmerlynn** [comment]
Tool, merging.

**Comment by thockin** [comment]
LGTM

**Pull request merged**